util/config/make_sys_basic_types.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env python
+
+"""Build SysCTypes.chpl module.
+
+Determines the size of various C types (e.g. long, size_t, etc) and creates a
+Chapel module with 'extern types' declaration so Chapel code can refer to the
+types.
+"""
+
+from __future__ import print_function
+
+import contextlib
+import logging
+import optparse
+import os
+import os.path
+import re
+import sys
+
+
+# List of C types. The items in each element are:
+#
+#  * C macro for maximum value for type
+#  * Name of Chapel type
+#  * Name of C type
+_types = [
+    ('INT_MAX', 'c_int', 'int'),
+    ('UINT_MAX', 'c_uint', 'uint'),
+    ('LONG_MAX', 'c_long', 'long'),
+    ('ULONG_MAX', 'c_ulong', 'unsigned long'),
+    ('LLONG_MAX', 'c_longlong', 'long long'),
+    ('ULLONG_MAX', 'c_ulonglong', 'unsigned long long'),
+    ('CHAR_MAX', 'c_char', 'char'),
+    ('SCHAR_MAX', 'c_schar', 'signed char'),
+    ('UCHAR_MAX', 'c_uchar', 'unsigned char'),
+    ('SHRT_MAX', 'c_short', 'short'),
+    ('USHRT_MAX', 'c_ushort', 'unsigned short'),
+    ('INTPTR_MAX', 'c_intptr', 'intptr_t'),
+    ('UINTPTR_MAX', 'c_uintptr', 'uintptr_t'),
+    ('PTRDIFF_MAX', 'c_ptrdiff', 'ptrdiff_t'),
+    ('SSIZE_MAX', 'ssize_t', 'ssize_t'),
+    ('SIZE_MAX', 'size_t', 'size_t'),
+]
+
+# Map of max values to chapel types.
+_max_value_to_chpl_type = {
+    '127': 'int(8)',
+    '255': 'uint(8)',
+    '32767': 'int(16)',
+    '65535': 'uint(16)',
+    '2147483647': 'int(32)',
+    '4294967295': 'uint(32)',
+    '9223372036854775807': 'int(64)',
+    '18446744073709551615': 'uint(64)',
+}
+
+
+def main():
+    """Parse command line arguments, create module, and print to stdout."""
+    args = _parse_args()
+    _setup_logging(args.verbose)
+    print(get_sys_c_types(args.doc))
+
+
+def get_sys_c_types(docs=False):
+    """Returns a string with the SysCTypes.chpl module content."""
+
+    # Find the $CHPL_HOME/util/config/ dir.
+    util_cfg_dir = os.path.abspath(os.path.dirname(__file__))
+    logging.debug('Util config dir: {0}'.format(util_cfg_dir))
+
+    # Get the C compile line.
+    compileline_cmd = os.path.join(util_cfg_dir, 'compileline')
+    cmd = '{compileline_cmd} --compile'.format(**locals())
+    compileline = os.popen(cmd).read().strip()
+    logging.debug('Compile line: {0}'.format(compileline))
+
+    # Create temp header file with *_MAX macros, then run it through the C
+    # preprocessor using the C compile line found above.
+    h_file = 'find_int_sizes_tmp.h'
+    with _ensure_deleted(h_file):
+        logging.debug('Creating temp header: {0}'.format(h_file))
+        with open(h_file, 'w') as fp:
+            fp.write("""
+#include "sys_basic.h"
+#include <limits.h>
+#include <stdint.h>
+#include <math.h>
+
+FIND_INT_SIZES_START
+""")
```

Comment:
Having this left-justified here really breaks up the flow, and since Python
uses indentation for semantics, makes it harder to read. Could you
move the whole C snippet to a global variable up above?